core/plugins/org.fusesource.ide.camel.model.service.impl/src/org/fusesource/ide/camel/model/service/internal/CamelModelPatcher.java

```diff
@@ -41,14 +38,9 @@ public static void applyVersionSpecificCatalogFixes(CamelCatalog catalog, CamelM
 			// can't work with a null value here - thats usually caused by a non existing catalog
 			return;
 		}
-		ComparableVersion loadedVersion = new ComparableVersion(camelVersion);
-		if (loadedVersion.compareTo(v217Ref) < 0) {
```

Comment:
by removing the version check, we may override some newly working syntax for some unknown version (even if chances are low)
It will also force to "loadmodel.getComponent" for all versions, maybe we can limit it to some range for performance sake (or maybe it is not worthy, performance impact is low)